java/ql/lib/semmle/code/java/regex/regex.qll

```diff
@@ -0,0 +1,933 @@
+import java
+private import RegexFlowConfigs
+
+// In all ranges handled by this library, `start` is inclusive and `end` is exclusive.
+/**
+ * A string literal that is used as a regular expression.
+ */
+abstract class RegexString extends StringLiteral {
+  /** Holds if a character set starts between `start` and `end`, including any negation character (`^`). */
```

Comment:
Terminology should be fixed throughout: "character class" vs. "character set". I believe the former is the usual name.